### PR TITLE
docs: explain missing file loading step for schema mapping

### PR DESCRIPTION
**Pull request by wgao19**
## Description

Explain the file transformer dependency and plugin setup for schema mapping.

## Related Issues

Addresses #13134

### CODE REVIEW

File: docs/docs/gatsby-config.md [hunk 1]
```diff
@@ -115,6 +115,8 @@ module.exports = {
 }
 ```
 
+You may need to install the appropriate file transformer and source plugins for Gatsby to pick up the mapping files. This applies to other file types later mentioned in this segment as well.
```
Comment (marcysutton) [inline docs/docs/gatsby-config.md]:
```suggestion
You may need to install the appropriate file transformer and source plugins for Gatsby to pick up the mapping files. This applies to other file types later mentioned in this segment as well. For example, to handle YAML files, you would install [`gatsby-transformer-yaml`](/packages/gatsby-transformer-yaml/?=yaml). 

> _Note: Additional source plugins can be found in the [plugin library](/plugins?=gatsby-source)._
```

What do you think about adding some links to common source plugins?

Comment (wgao19) [reply docs/docs/gatsby-config.md]: Hey @marcysutton, I added a link to the YAML transformer plugin and rephrased a little bit. How do you think it reads now?

File: docs/docs/gatsby-config.md [hunk 2]
```diff
@@ -115,6 +115,8 @@ module.exports = {
 }
 ```
 
+You may need to install the appropriate file transformer (in this case [YAML](https://www.gatsbyjs.org/packages/gatsby-transformer-yaml/)) and set up [gatsby-source-filesystem](https://www.gatsbyjs.org/packages/gatsby-source-filesystem/) properly for Gatsby to pick up the mapping files. This applies to other file types later mentioned in this segment as well.
```
Comment (marcysutton) [inline docs/docs/gatsby-config.md]:
```suggestion
You may need to install the appropriate file transformer (in this case [YAML](/packages/gatsby-transformer-yaml/)) and set up [gatsby-source-filesystem](/packages/gatsby-source-filesystem/) properly for Gatsby to pick up the mapping files. This applies to other file types later mentioned in this segment as well.
```
We should keep these URLs relative to the docs site. But that's it! Looking great!

Comment (wgao19) [reply docs/docs/gatsby-config.md]: @marcysutton updated :)))))

**Pull request merged**